src/kbmod/region_search.py

```diff
@@ -1,424 +1,659 @@
-                    collections=collections,
-                    butler=self.new_butler(),
-                    trim_uri_func=trim_uri_func,
-                )
-                for chunk in data_id_chunks
-            ]
-            for future in as_completed(futures):
-                uris.extend(future.result())
+        if len(ic_to_export) < 1:
+            raise ValueError(f"ImageCollection is empty, cannot export {ic_to_export}")
 
-        return uris
+        new_ic = ic_to_export.copy()
 
-    def get_center_ra_dec(self, region):
+        # Add the metadata about the guess distance used when choosing this ImageCollection
+        if guess_dist is not None:
+            new_ic.data["helio_guess_dist"] = guess_dist
+
+        # Add the metadata about the patch this ImageCollection represents.
+        if patch is not None:
+            if not isinstance(patch, Patch):
+                if not isinstance(patch, int):
+                    raise ValueError("Patch must be an integer or a Patch object")
+                patch = self.get_patches()[patch]
+
+            # Populate the patch information to construct the patch WCS
+            patch_wcs = patch.to_wcs()
+            new_ic.data["global_wcs"] = patch_wcs.to_header_string()
+            new_ic.data["global_wcs_pixel_shape_0"] = patch_wcs.pixel_shape[0]
+            new_ic.data["global_wcs_pixel_shape_1"] = patch_wcs.pixel_shape[1]
+
+        # Reset standardizer-related metadata in our ImageCollection.
+        new_ic.meta["n_stds"] = len(new_ic)
+        new_ic.data["std_idx"] = range(len(new_ic))
+
+        return new_ic
+
+    def get_image_collection_from_patch(self, patch, guess_dist=0.0, min_overlap=0, max_images=None):
         """
-        Get the center RA and Dec for the given region.
+        Filters down an ImageCollection to all images that overlap with a given patch.
+
+        A patch may be specified by a Patch object or by its index in the PatchGrid.
+
+        Adds an 'overlap_deg' column to the ImageCollection, which is the area of overlap
+        between the patch and each image in square degrees.
+
+        Parameters
+        ----------
+        patch : Patch or int
+            The patch to filter by.
+        guess_dist : float, optional
+            The guess distance to use for reflex correction. If 0.0, the original coordinates are used.
+        min_overlap : float, optional
+            The minimum overlap area in square degrees to include an image in the filtered ImageCollection.
+        max_images : int, optional
+            The maximum number of images to return. If None, all images are returned.
+        """
+        if guess_dist is not None and guess_dist != 0.0 and guess_dist not in self.guess_dists:
+            raise ValueError(f"Guess distance {guess_dist} not specified for Region Search")
+
+        if not isinstance(patch, Patch):
+            if not (isinstance(patch, int) or isinstance(patch, np.integer)):
```

Comment:
I would suggest using this same conditional logic on line 455, 456. I think you can even do this:
```
if not isinstance(patch, [int, np.integer]):
```

And on line 455, 456, `if not isinstnace(patch, [Patch, int]):`